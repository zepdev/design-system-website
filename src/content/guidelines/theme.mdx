---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
