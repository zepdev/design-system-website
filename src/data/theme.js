module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
