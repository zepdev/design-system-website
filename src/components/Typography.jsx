import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import theme from '../data/theme'
import withStyles from 'react-jss'

const styles = theme => ({
  table: {
    width: '100%',
    marginBottom: theme.spacing.xl * 2,
    borderCollapse: 'collapse',
    overflowX: 'scroll',
  },
  th: {
    padding: `${ theme.spacing.m }px ${ theme.spacing.l }px`,
    textAlign: 'left',
    display: 'table-cell',
  },
  tr: {
    display: 'table-row',
  },
  tbody: {
    display: 'table-row-group',
  },
  header: {
    background: theme.colors.gray.grayLightest.hex,
    display: 'table-header-group',
  },
  lowercase: {
    textTransform: 'lowercase',
  },
  capitalize: {
    textTransform: 'capitalize',
  },
  borderBottom: {
    borderBottom: `1px solid ${ theme.colors.gray.grayLighter.hex }`,
  },
})

const Typography = ({ section, classes }) => {
  const header =
    section === 'basics'
      ? ['specification', 'px', 'rem', 'weight']
      : ['name', 'rem', 'px', 'specifications', 'actual size']
  const content =
    section === 'basics' ? theme.typography.basics : theme.typography.headings[section]

  return (
    <table className={classes.table}>
      <caption className="visually-hidden">{`${ section } typography variations`}</caption>
      <thead className={classes.header}>
        <tr className={classes.tr}>
          {header.map(elem => (
            <th
              scope="col"
              className={classnames(classes.th, 'zep-typo--normal-8', {
                [classes.lowercase]: elem === 'rem' || elem === 'px',
                [classes.capitalize]: elem !== 'rem' && elem !== 'px',
              })}
              key={`header_${ elem }`}
            >
              {elem}
            </th>
          ))}
        </tr>
      </thead>
      <tbody className={classes.tbody}>
        {Object.keys(content).map(elem => (
          <tr key={`row_${ elem }`} className={classes.tr}>
            {header.map((item, idx) => (
              <td
                className={classnames(
                  classes.th,
                  classes.lowercase,
                  classes.borderBottom,
                  item !== 'actual size' ? 'zep-typo--special-1' : `${ elem }`
                )}
                key={`typography${ idx }`}
                scope={idx === 0 ? 'row' : null}
              >
                {item === 'actual size' ? (
                  'Lorem Ipsum'
                ) : item === 'name' ? (
                  elem
                ) : item === 'specifications' ? (
                  <span>
                    {content[elem][item].map(p => (
                      <p key={`specification_${ p }`} className="zep-typo--normal-8">
                        {p}
                      </p>
                    ))}
                  </span>
                ) : content[elem][item] ? (
                  content[elem][item]
                ) : (
                  '-'
                )}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  )
}

Typography.propTypes = {
  classes: PropTypes.object.isRequired,
  section: PropTypes.string.isRequired,
}

export default withStyles(styles)(Typography)
