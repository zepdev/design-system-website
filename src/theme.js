import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
