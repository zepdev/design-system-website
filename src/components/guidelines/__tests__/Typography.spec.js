import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'react-jss'
import theme from 'zeppelin-element-library/bundle/themes/theme.json'
import Typography from '../Typography'

describe('Typography', () => {
  it('renders correctly', () => {
    const mockProps = {
      section: 'basics',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <Typography {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders the a list of p for specifications', () => {
    const mockProps = {
      section: 'normal',
    }
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Typography {...mockProps} />
      </ThemeProvider>
    )
    const span = getByTestId('specificationszep-typo--normal-body1')
    expect(span.children).toHaveLength(5)
  })
})
