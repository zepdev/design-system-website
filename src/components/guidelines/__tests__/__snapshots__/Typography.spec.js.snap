// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Typography-root-0-1-1"
      >
        <table
          class="Typography-table-0-1-2"
        >
          <caption
            class="zep-visually-hidden"
          >
            basics typography variations
          </caption>
          <thead
            class="Typography-header-0-1-6"
          >
            <tr
              class="Typography-tr-0-1-4"
            >
              <th
                class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-capitalize-0-1-8"
                scope="col"
              >
                specification
              </th>
              <th
                class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="col"
              >
                px
              </th>
              <th
                class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="col"
              >
                rem
              </th>
              <th
                class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-capitalize-0-1-8"
                scope="col"
              >
                weight
              </th>
            </tr>
          </thead>
          <tbody
            class="Typography-tbody-0-1-5"
          >
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-base
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-base
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-base
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                line-height-base
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-weight-base
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-xs
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-sm
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
            <tr
              class="Typography-tr-0-1-4"
            >
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
                scope="row"
              >
                font-size-lg
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
              <td
                class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              >
                -
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Typography-root-0-1-1"
    >
      <table
        class="Typography-table-0-1-2"
      >
        <caption
          class="zep-visually-hidden"
        >
          basics typography variations
        </caption>
        <thead
          class="Typography-header-0-1-6"
        >
          <tr
            class="Typography-tr-0-1-4"
          >
            <th
              class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-capitalize-0-1-8"
              scope="col"
            >
              specification
            </th>
            <th
              class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="col"
            >
              px
            </th>
            <th
              class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="col"
            >
              rem
            </th>
            <th
              class="Typography-th-0-1-3 zep-typo--normal-body1 Typography-capitalize-0-1-8"
              scope="col"
            >
              weight
            </th>
          </tr>
        </thead>
        <tbody
          class="Typography-tbody-0-1-5"
        >
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-base
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-base
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-base
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              line-height-base
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-weight-base
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-xs
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-sm
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
          <tr
            class="Typography-tr-0-1-4"
          >
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
              scope="row"
            >
              font-size-lg
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
            <td
              class="Typography-th-0-1-3 Typography-borderBottom-0-1-9 zep-typo--normal-body1 Typography-lowercase-0-1-7"
            >
              -
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
